File: Week1/Projects/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

//"initializing" functions
int get_size(void);
void print_pyramid(int height);

//"control center"
int main(void)
{
    int height = get_size();
    print_pyramid(height);
}

//Function to get
int get_size(void)
{
    int height;
    do
    {
        height = get_int("Size: ");
    }
    while (height < 1 || height > 8);
    return height;
}

//Function to print
void print_pyramid(int height)
{
    //Counter set to 0.. as long as counter is less than requested height, keep counting (aka doing whats in the loop)
    for (int row = 0; row < height; row++)
    {
        //another counter for spaces before hashes
        for (int spaces = 0; spaces < height - row - 1; spaces++)
        {
            printf(" ");
        }
        //Left Columns
        for (int LColumn = 0; LColumn <= row; LColumn++)
        {
            printf("#");
        }
        //print gap
        printf("  ");
        //Right columns
        for (int RColumn = 0; RColumn <= row; RColumn++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: Week6/Projects/mario.py

while True:
    try:
        height = int(input("height: "))
        if 0 < height <= 8:
            break
    except ValueError:
        pass


row = 0
while row < height:
    spaces = 0
    while spaces < height - row - 1:
        print(" ", end="")
        spaces += 1
    i = 0
    while i <= row:
        print("#", end="")
        i += 1
    print("  ", end="")
    j = 0
    while j <= row:
        print("#", end="")
        j += 1
    print()
    row += 1
